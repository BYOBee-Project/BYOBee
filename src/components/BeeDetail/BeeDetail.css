.bee-detail {
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.bee-detail-title {
  text-align: center;
  margin: 5vh 0 0 0;
  padding: 0;
}

.bee-detail-links {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2vh 0 4vh 0;
}

.bee-details {
  display: flex;
  flex-direction: column;
}

.bee-detail-label {
  font-weight: bold;
}

.bee-detail-images {
  display: flex;
  gap: 2vw;
}

.bee-detail-image {
  width: 25vw;
  margin: 0 0 2vh 0;
  border: 1px solid black;
}
