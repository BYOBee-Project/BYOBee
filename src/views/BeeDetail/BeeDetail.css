.bee-detail {
  width: 65%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 5px;
  background-color: hsla(4, 82%, 67%, 0.827);
  border-radius: 10px;
  margin-top: 5vh;
  padding: 2vh 4vw 2vh 4vw;
}

.bee-detail-title {
  font-family: 'Londrina Solid', cursive;
  font-size: 4vh;
  text-align: center;
  padding: 0;
}

.bee-detail-links {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Link {
  color: white;
}

.bee-details {
  display: flex;
  flex-direction: column;
  margin: 0;
}

.bee-detail-label {
  font-weight: bold;
}

.bee-detail-images {
  display: flex;
  gap: 2vw;
  filter: drop-shadow(1px 1px 2px #000007);
}

.bee-detail-image {
  width: 25vw;
  margin: 0 0 2vh 0;
  border: 1px solid black;
  border-radius: 10px;
}
