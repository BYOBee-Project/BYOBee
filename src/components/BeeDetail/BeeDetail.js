import React from 'react';
import './BeeDetail.css';

export default function BeeDetail() {
  return <div className="BeeDetail">BeeDetail</div>;
}
