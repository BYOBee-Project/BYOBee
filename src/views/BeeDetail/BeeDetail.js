import React from 'react';
import './BeeDetail.css';
import { findBeeById } from '../../services/bees';
import { useState, useEffect } from 'react';
import { Link, useParams } from 'react-router-dom';

export default function BeeDetail({ currentUser }) {
  const params = useParams();
  const [currentBee, setCurrentBee] = useState({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchBee = async () => {
      const data = await findBeeById(params.id);
      setCurrentBee(data);
      setLoading(false);
    };
    fetchBee();
  }, [params.id]);

  if (loading) return <h1 className="loader">Loading...</h1>;
  return (
    <div className="bee-detail">
      <h1 className="bee-detail-title">{currentBee.name}</h1>
      <div className="bee-detail-links">
        {!currentUser && (
          <button>
            <Link to={`/auth`} className="bee-card-link">
              + Add to Collection
            </Link>
          </button>
        )}
        {currentUser && (
          <button>
            <Link to={`/beeform/${params.id}`} className="bee-card-link">
              + Add to Collection
            </Link>
          </button>
        )}
      </div>
      <div className="bee-detail-images">
        <img className="bee-detail-image" src={currentBee.image1} alt={currentBee.name} />
        <img className="bee-detail-image" src={currentBee.image2} alt={currentBee.name} />
      </div>

      <div className="bee-details">
        <p className="bee-detail-info">
          <span className="bee-detail-label">Species: </span>
          {currentBee.species}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Family: </span>
          {currentBee.family}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Region: </span>
          {currentBee.region}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Description: </span>
          {currentBee.description}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Preferred Forage: </span>
          {currentBee.forage}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Nesting Habits: </span>
          {currentBee.habitat}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Active Season: </span>
          {currentBee.season}
        </p>
        <p className="bee-detail-info">
          <span className="bee-detail-label">Fun Fact: </span>
          {currentBee.fact}
        </p>
      </div>
      <Link to={`/`} className="bee-detail-link">
        Back to Home
      </Link>
    </div>
  );
}
