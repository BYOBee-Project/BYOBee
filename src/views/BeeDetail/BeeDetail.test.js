import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import BeeDetail from './BeeDetail.js';

test('BeeDetail displays Bee detail information', async () => {
  const container = render(
    <MemoryRouter initialEntries={['/bees/6']}>
      <Route exact path="/bees/:id">
        <BeeDetail currentUser={{ user: { id: 'd1beb9ec-d137-4e7f-ac5c-4c9197075a15' } }} />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Alfalfa Leafcutter Bee');
  expect(container).toMatchSnapshot();
});
