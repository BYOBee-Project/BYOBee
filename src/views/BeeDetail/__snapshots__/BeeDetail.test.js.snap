// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeeDetail displays Bee detail information 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bee-detail"
      >
        <h1
          class="bee-detail-title"
        >
          Alfalfa Leafcutter Bee
        </h1>
        <div
          class="bee-detail-images"
        >
          <img
            alt="Alfalfa Leafcutter Bee"
            class="bee-detail-image"
            src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-1.jpg"
          />
          <img
            alt="Alfalfa Leafcutter Bee"
            class="bee-detail-image"
            src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-2.jpg"
          />
        </div>
        <div
          class="bee-detail-links"
        >
          <button>
            <a
              class="bee-card-link"
              href="/beeform/6"
            >
              + Add to Collection
            </a>
          </button>
        </div>
        <div
          class="bee-details"
        >
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Species: 
            </span>
            Megachile rotundata
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Family: 
            </span>
            Megachilidae
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Region: 
            </span>
            Introduced European species that is now common across North America
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Description: 
            </span>
            Female alfalfa leafcutter bees are mostly black, and are fairly smooth when viewed from above. The have slightly pointed abdomens and carry pollen on special velcro-like hairs under their abdomen (called scopa). The males are lighter in color and slightly fuzzier with more prominent striping on their rounded abdomens. They also have strikingly green-tinted eyes!
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Preferred Forage: 
            </span>
            Clovers, legumes, rapeseed, and… alfalfa, of course!
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Nesting Habits: 
            </span>
            M. rotundata are solitary, tunnel-nesting bees that will reuse holes drilled by other insects or animals, as well as hollow reeds and flower stems to make their nest cavities.  Humans can provide nesting sites via cardboard tubes, natural reeds, or wooden nesting trays (similar to mason bees). These little bees prefer holes with a 6mm diameter! 
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Active Season: 
            </span>
            June through September in northern states, with longer active seasons in the south.
          </p>
          <p
            class="bee-detail-info"
          >
            <span
              class="bee-detail-label"
            >
              Fun Fact: 
            </span>
            These bees were purposefully introduced by the federal government in the 1940s to increase pollination efforts of alfalfa and yield higher seed production. You can thank these bees for all the protein-rich alfalfa that we feed to dairy cattle for yummy foods like cheese and ice cream!
          </p>
        </div>
        <button>
          <a
            class="bee-detail-link"
            href="/"
          >
            Back to Home
          </a>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bee-detail"
    >
      <h1
        class="bee-detail-title"
      >
        Alfalfa Leafcutter Bee
      </h1>
      <div
        class="bee-detail-images"
      >
        <img
          alt="Alfalfa Leafcutter Bee"
          class="bee-detail-image"
          src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-1.jpg"
        />
        <img
          alt="Alfalfa Leafcutter Bee"
          class="bee-detail-image"
          src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-2.jpg"
        />
      </div>
      <div
        class="bee-detail-links"
      >
        <button>
          <a
            class="bee-card-link"
            href="/beeform/6"
          >
            + Add to Collection
          </a>
        </button>
      </div>
      <div
        class="bee-details"
      >
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Species: 
          </span>
          Megachile rotundata
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Family: 
          </span>
          Megachilidae
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Region: 
          </span>
          Introduced European species that is now common across North America
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Description: 
          </span>
          Female alfalfa leafcutter bees are mostly black, and are fairly smooth when viewed from above. The have slightly pointed abdomens and carry pollen on special velcro-like hairs under their abdomen (called scopa). The males are lighter in color and slightly fuzzier with more prominent striping on their rounded abdomens. They also have strikingly green-tinted eyes!
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Preferred Forage: 
          </span>
          Clovers, legumes, rapeseed, and… alfalfa, of course!
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Nesting Habits: 
          </span>
          M. rotundata are solitary, tunnel-nesting bees that will reuse holes drilled by other insects or animals, as well as hollow reeds and flower stems to make their nest cavities.  Humans can provide nesting sites via cardboard tubes, natural reeds, or wooden nesting trays (similar to mason bees). These little bees prefer holes with a 6mm diameter! 
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Active Season: 
          </span>
          June through September in northern states, with longer active seasons in the south.
        </p>
        <p
          class="bee-detail-info"
        >
          <span
            class="bee-detail-label"
          >
            Fun Fact: 
          </span>
          These bees were purposefully introduced by the federal government in the 1940s to increase pollination efforts of alfalfa and yield higher seed production. You can thank these bees for all the protein-rich alfalfa that we feed to dairy cattle for yummy foods like cheese and ice cream!
        </p>
      </div>
      <button>
        <a
          class="bee-detail-link"
          href="/"
        >
          Back to Home
        </a>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
